src/test/java/nl/novi/catsittermanager/services/CatServiceTest.java

```diff
@@ -85,22 +88,51 @@ void testGetCat_shouldFetchCatWithSpecificId() {
         assertEquals(expectedCatDto, actualCatDto);
     }
 
+    @Test
+    void testGetCat_shouldFetchCatWithSpecificId_RecordNotFoundException() {
+        // Arrange
+        Customer owner1 = new Customer();
+        owner1.setUsername("marjet_bosma");
+
+        UUID catId = UUID.randomUUID();
+        Cat cat = new Cat(catId, "Firsa", LocalDate.of(2006, 7, 1), "V", "Europese Korthaar", "Vriendelijke, maar verlegen kat", true, true, "Dierenkliniek Zuilen", "030-2446933", "Thiafeline", "5mg 1dd", owner1);
+
+        CatDto expectedCatDto = new CatDto(catId, "Firsa", LocalDate.of(2006, 7, 1), "V", "Europese Korthaar", "Vriendelijke, maar verlegen kat", true, true, "Dierenkliniek Zuilen", "030-2446933", "Thiafeline", "5mg 1dd", owner1.getUsername());
+
+        // Mock behavior of the repository
+//        when(catRepos.findById(catId)).thenThrow(new RecordNotFoundException(HttpStatus.NOT_FOUND, "No cat found with this id."));
+        when(catRepos.findById(catId)).thenReturn(Optional.empty());
+
+        // Act
+        RecordNotFoundException result = assertThrows(RecordNotFoundException.class, () -> catService.getCat(catId));
+
+        // Assert
+        assertEquals("No cat found with this id.", result.getMessage());
+    }
+
     @Test
     void testCreateCat_newCatObjectShouldBeCreated_fieldsInCatInputDtoAndCatDtoShouldContainTheSameValues() {
 
         //Arrange
         Customer owner = new Customer();
         owner.setUsername("marjet_bosma");
+        Cat expectedCat = new Cat();
+        expectedCat.setId(UUID.randomUUID());
 
         when(customerRepos.findById(owner.getUsername())).thenReturn(Optional.of(owner));
```

Comment:
Is the username the primary key here? I would expect the findById to take a Id parameter but you are passing the username here?

Reply:
Yes, the username is used as the id and primary key. I found that the most convenient, especially with the login system.